src/hugr/validate.rs

```diff
@@ -27,6 +28,8 @@ struct ValidationContext<'a> {
     hugr: &'a Hugr,
     /// Dominator tree for each CFG region, using the container node as index.
     dominators: HashMap<Node, DominatorTree>,
+    /// Resource requirements associated with each edge
+    resources: HashMap<PortIndex, ResourceSet>,
```

Comment:
In the Hugr level we normally address ports by `(Node, Port)` instead of the internal portgraph's `NodeIndex`. Maybe using that would avoid some extra queries?

Reply:
Yes, that's much nicer!